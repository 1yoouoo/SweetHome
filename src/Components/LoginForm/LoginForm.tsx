import StyledButton from "../../Views/StyledButton/StyledButton";
import Logo from "../../Views/Logo/Logo";
import "./LoginForm.scss";
import InputBox from "../../Views/InputBox/InputBox";
import { useRef } from "react";
import API from "../../API/API";
// type
const LoginForm = () => {
  const emailRef = useRef<HTMLInputElement>(null);
  const passwordRef = useRef<HTMLInputElement>(null);
  const onSubmit = async (e: React.SyntheticEvent) => {
    e.preventDefault();
    const response = await API.logIn({
      email: emailRef.current?.value,
      password: passwordRef.current?.value,
    });
    if (response?.data.error === null) {
      const token = response?.headers.token;
      response.data.data.userLoginResponse;
      localStorage.setItem("token", token);
      console.log(response);
      window.location.reload();
    } else {
      alert(response.data.error.message);
    }
  };
  return (
    <>
      <form className="LoginForm" onSubmit={onSubmit}>
        <Logo width="175px" height="51px" />
        <section className="LoginForm__input-section">
          <InputBox placeholder="Email" inputRef={emailRef} />
          <InputBox
            placeholder="Password"
            inputRef={passwordRef}
            type="password"
          />
          <div style={{ margin: "20px 0" }}>
            <StyledButton
              width="258px"
              height="34px"
              backgroundColor="#0095f6"
              content="Log In"
              color="#fff"
            />
          </div>
        </section>
        <div className="LoginForm__division">
          <span className="LoginForm__division--left"></span>
          <span className="LoginForm__division--or">OR</span>
          <span className="LoginForm__division--right"></span>
        </div>
        <div className="LoginForm__forgot-pw">Forgot password ?</div>
      </form>
    </>
  );
};
export default LoginForm;
