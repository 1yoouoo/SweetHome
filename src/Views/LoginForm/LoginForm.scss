.LoginForm {
  display: flex;
  flex-direction: column;
  background-color: rebeccapurple;
  align-items: center;
}
