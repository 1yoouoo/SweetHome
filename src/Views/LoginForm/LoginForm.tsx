import StyledButton from "../StyledButton/StyledButton";
import Logo from "../Logo/Logo";
import "./LoginForm.scss";
import InputBox from "../InputBox/InputBox";
import { useState } from "react";
// type
interface inputValueType {
  email: string;
  password: string;
}
const LoginForm = () => {
  const [inputValue, setInputValue] = useState<inputValueType>({
    email: "",
    password: "",
  });
  const onChangeValue = (e: React.ChangeEvent<HTMLInputElement>) => {
    setInputValue({
      ...inputValue,
      [e.currentTarget.name]: e.currentTarget.value,
    });
  };
  const onSubmit = (e: React.SyntheticEvent) => {
    e.preventDefault();
    console.log(inputValue);
  };
  return (
    <>
      <form className="LoginForm" onSubmit={onSubmit}>
        <Logo width="175px" height="51px" />
        <section className="LoginForm__input-section">
          <InputBox
            placeholder="Email"
            name="email"
            value={inputValue.email}
            onChangeValue={onChangeValue}
          />
          <InputBox
            placeholder="Password"
            name="password"
            value={inputValue.password}
            onChangeValue={onChangeValue}
          />
          <div style={{ margin: "20px 0" }}>
            <StyledButton
              width="258px"
              height="34px"
              backgroundColor="#0095f6"
              content="Log In"
              color="#fff"
            />
          </div>
        </section>
        <div className="LoginForm__division">
          <span className="LoginForm__division--left"></span>
          <span className="LoginForm__division--or">OR</span>
          <span className="LoginForm__division--right"></span>
        </div>
        <div className="LoginForm__forgot-pw">Forgot password ?</div>
      </form>
    </>
  );
};
export default LoginForm;
