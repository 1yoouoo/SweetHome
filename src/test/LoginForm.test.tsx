import * as React from "react";
import { fireEvent, render } from "@testing-library/react";
import LoginForm from "../Components/LoginForm/LoginForm";

describe("<LoginForm /> test", () => {
  it("matches snapshot", () => {
    const utils = render(<LoginForm />);
    expect(utils.container).toMatchSnapshot(); // snapshot match
  });
  // test("test child component", () => {
  //   const { getByText } = render(<LoginForm />);
  //   expect(getByText(/Logo/i)).toBeInTheDocument();
  // });

  // it("screen test", () => {
  //   const utils = render(<LoginForm />);

  //   const h2 = utils.container.querySelectorAll("h2"); // h2태그 DOM 가져오기
  //   console.log(h2);
  //   h2 && expect(h2[0].innerHTML).toBe("Hello. Jest!!"); // h2의 innerHTML이 Hello, Jest!! 인지 테스트
  //   h2 && expect(h2[1].innerHTML).toBe("hihihia"); // h2의 innerHTML이 Hello, Jest!! 인지 테스트
  // });

  // it("button test", () => {
  //   const utils = render(<LoginForm />);

  //   const str = utils.getByText("number: 0"); // 텍스트가 number: 0인 DOM 가져오기
  //   const increaseButton = utils.getByText("증가"); // 텍스트가 증가인 DOM 가져오기
  //   const decreaseButton = utils.getByText("감소"); // 텍스트가 감소인 DOM 가져오기

  //   fireEvent.click(increaseButton); // 증가버튼 클릭이벤트 실행
  //   fireEvent.click(increaseButton); // 증가버튼 클릭이벤트 실행
  //   expect(str).toHaveTextContent("number: 6"); // str의 텍스트가 number: 6인지 확인

  //   fireEvent.click(decreaseButton); // 감소버튼 클릭이벤트 실행
  //   fireEvent.click(decreaseButton); // 감소버튼 클릭이벤트 실행
  //   expect(str).toHaveTextContent("number: 2"); // str의 텍스트가 number: 2인지 확인
  // });
});
