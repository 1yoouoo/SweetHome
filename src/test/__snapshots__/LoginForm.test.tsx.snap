// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> test matches snapshot 1`] = `
<div>
  <form
    class="LoginForm"
  />
</div>
`;
